build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/extractor/GoVersionUtils.java

```diff
@@ -97,12 +98,27 @@ public static boolean isCompatibleGoModuleNaming(String projectName, String vers
      */
     public static String getSubModule(String projectName) {
         if (StringUtils.isBlank(projectName)) {
-            return null;
+            return StringUtils.EMPTY;
         }
         String[] parts = projectName.split("/", 4);
         if (parts.length >= 4) {
             return parts[3];
         }
-        return null;
+        return StringUtils.EMPTY;
+    }
+
+    /**
+     * @param path A file path
+     * @return Parent path of the input path as if it was a file. Empty string if the path has no parent.
+     */
+    public static String getParent(String path) {
+        if (StringUtils.isEmpty(path)) {
+            return StringUtils.EMPTY;
+        }
+        String parentPath = new File(path).getParent();
+        if (StringUtils.isNotEmpty(parentPath)) {
+            return parentPath.replace('\\', '/');
+        }
+        return StringUtils.EMPTY;
```

Comment:
The above if statement seems to be redundant. The following code:
```
        if (StringUtils.isNotEmpty(parentPath)) {
            return parentPath.replace('\\', '/');
        }
        return StringUtils.EMPTY;
```
can be reduced to:
```
return parentPath.replace('\\', '/');
```

Reply:
But parentPath can be null

Reply:
You're correct!